src/test/java/org/opensearch/security/auditlog/integration/TestAuditlogImpl.java

```diff
@@ -25,23 +28,61 @@
 
 public class TestAuditlogImpl extends AuditLogSink {
 
+    /** Use the results of `doThenWaitForMessages(...)` instead */
+    @Deprecated
     public static List<AuditMessage> messages = new ArrayList<AuditMessage>(100);
+    /** Check messages indvidually instead of searching this string */
+    @Deprecated
     public static StringBuffer sb = new StringBuffer();
+    private static final AtomicReference<CountDownLatch> countDownRef = new AtomicReference<>();
+    private static final AtomicReference<List<AuditMessage>> messagesRef = new AtomicReference<>();
 
     public TestAuditlogImpl(String name, Settings settings, String settingsPrefix, AuditLogSink fallbackSink) {
         super(name, settings, null, fallbackSink);
     }
 
-
-    public synchronized boolean  doStore(AuditMessage msg) {
+    public synchronized boolean doStore(AuditMessage msg) {
+        if (messagesRef.get() == null || countDownRef.get() == null) {
+            throw new RuntimeException("No message latch is waiting");
+        }
         sb.append(msg.toPrettyString()+System.lineSeparator());
-        messages.add(msg);
+        messagesRef.get().add(msg);
+        countDownRef.get().countDown();
         return true;
     }
 
+    /** Unneeded after switching to `doThenWaitForMessages(...)` as data is automatically flushed */
+    @Deprecated
```

Comment:
Is there any reason for keeping 2 different methods for the exactly same purpose? If not, replacing the old method with the new method would retain consistency.

Reply:
The testRestMethod has been the source of trouble, lets incrementally invest in the space.

If none of these other tests need to be updated, it will save us time/trouble. Each assert that was updated required investigation to confirm the correct behavior.  With  the api deprecation, this will help ensure new tests use the more reliable technique.